application/models/services/CoreQuestionAttributeProvider.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+namespace LimeSurvey\Models\Services;
+
+/**
+ * Provides question attribute definitions from question types
+ */
+
+class CoreQuestionAttributeProvider extends QuestionAttributeProvider
+{
+    /** @inheritdoc */
+    public function getDefinitions($options = [])
+    {
+        /** @var string question type */
+        $questionType = self::getQuestionType($options);
+        if (empty($questionType)) {
+            return [];
+        }
+
+        /** @var boolean */
+        $advancedOnly = !empty($options['advancedOnly']);
+
+        return $this->getQuestionAttributes($questionType, $advancedOnly);
+    }
+
+    /**
+     * Return the question attribute settings for the passed type (parameter)
+     *
+     * @param string $questionType : type of question (this is the attribute 'question_type' in table question_theme)
+     * @param boolean $advancedOnly If true, only fetch advanced attributes
+     * @return array<string,array> the attribute settings for this question type
+     *                 returns values from getGeneralAttributesFromXml and getAdvancedAttributesFromXml if this fails
+     *                 getAttributesDefinition and getDefaultSettings are returned
+     *
+     * @throws \CException
+     */
+    protected function getQuestionAttributes($questionType, $advancedOnly = false)
+    {
+        $xmlFilePath = \QuestionTheme::getQuestionXMLPathForBaseType($questionType);
+        if ($advancedOnly) {
+            $generalAttributes = [];
+        } else {
+            // Get attributes from config.xml
+            $generalAttributes = $this->getGeneralAttibutesFromXml($xmlFilePath);
+        }
+        $advancedAttributes = $this->getAdvancedAttributesFromXml($xmlFilePath);
+
+        /** @var array<string,array> An array of question attributes */
+        $attributes = array_merge($generalAttributes, $advancedAttributes);
+
+        return $attributes;
+    }
+
+    /**
+     * Read question attributes from XML file and convert it to array
+     *
+     * @param string $xmlFilePath Path to XML
+     *
+     * @return array<string,array> The general attribute settings for this question type
+     */
+    protected function getGeneralAttibutesFromXml($xmlFilePath)
+    {
+        /** @var array<string,array> An array of question attributes */
+        $attributes = [];
+
+        if (file_exists($xmlFilePath)) {
+            // load xml file
+            if (\PHP_VERSION_ID < 80000) {
+                libxml_disable_entity_loader(false);
+            }
+            $xmlConfig = simplexml_load_file($xmlFilePath);
```

Comment:
How about factor out the simplexml_load_file to a separate method, together with disable entity loader and json decode/encode thingy? Some code duplication present now.

Reply:
Do you mean creating something like an xml reader service?

Reply:
I agree, but maybe on another story? I just want to wrap this. The later we merge, more chances of conflict. :)

Reply:
> Do you mean creating something like an xml reader service?

Just a simple function wrapping those lines would be enough.

Reply:
> I just want to wrap this

I've heard that before, too many times (not from you). :) The end result: On to the next thing, and the next thing, and the next...

Reply:
Wait, I'm retarded. I only wrote this some year ago: https://github.com/LimeSurvey/LimeSurvey/blob/master/application/models/ExtensionConfig.php#L149

Reply:
Ready for your re-review